E/video generator

chufucious opened the pull request:
### Overview

Adds Remotion to the web project in `pages/create-video`. For now the first video is for Bills, and has hardcoded scenes "Title", "Summary", and "Last Votes".

### Tasks
- [x] Scene "Title" 
- [x] Scene "Summary"
- [x] Scene "Last Votes"
- [x] Scene "Sponsors"
- [x] Scene "End Title"
- [x] Hook up scenes to dynamic data
- [x] Add animations to scenes (awaiting animations)
- [x] Add "Create Video" button on Bills page (needs basic UX + design)
- [ ] Build out "Create Video" page with video options
- [ ] Server rendering and download

chufucious commented on package.json:
```diff
@@ -18,6 +18,11 @@
     "@radix-ui/react-separator": "^1.0.3",
     "@radix-ui/react-tooltip": "^1.0.7",
+    "@remotion/cli": "^4.0.141",
+    "@remotion/google-fonts": "^4.0.141",
```
I realize I won't need google-fonts because I'm using what's already in the `web` project. Will remove this package later.

wileymc commented on pages/create-video/components/AlignEnd.tsx:
```diff
@@ -0,0 +1,11 @@
+  alignSelf: "flex-end",
+};
+
+export const AlignEnd: React.FC<{
+  children: React.ReactNode;
+}> = ({ children }) => {
+  return <div style={container}>{children}</div>;
+};
```
I don't see this component used anywhere is it necessary?

chufucious replied:
Will remove, was part of boilerplate.

wileymc commented on pages/create-video/components/Button/Button.tsx:
We already have a components/Button component.  Is this necessary? If it is, can you wrap our existing Button so that we don't duplicate styles and rename this export to avoid confusion?

chufucious replied:
Will remove, was part of boilerplate.

wileymc commented on pages/create-video/components/Button/styles.module.css:
Where are these --geist variables defined? Is this just boilerplate that can be removed?

wileymc commented on pages/create-video/remotion/MyComp/Main.tsx:
```diff
@@ -0,0 +1,104 @@
+  return (
+    <Sequence>
+      <div
+        style={{
+          display: "flex",
+          flexDirection: "column",
+          alignItems: "left",
+          justifyContent: "top",
+          padding: "8rem 4rem", // 120pt 40pt
+          gap: "4rem",
+        }}
```
Ideally we eliminate these inline styles in favor of using a SCSS module to keep the markup clean

chufucious replied:
Yup! Inlining for now for dev speed + iteration. Will move to module later.

wileymc commented on pages/create-video/remotion/MyComp/Main.tsx:
```diff
@@ -0,0 +1,104 @@
+// import { TextFade } from "./TextFade";
+// import { IssueTags } from "../../../../components/IssueTags/IssueTags";
+
+const tag: React.CSSProperties = {
+  width: "140pt",
+  height: "40pt",
+  padding: "24pt 14pt",
+  borderRadius: "32pt",
+  backgroundColor: "#003C55",
+  fontFamily: "proxima_nova",
+  fontSize: "2.25rem",
+  color: "white",
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+};
+
+const bigTag: React.CSSProperties = {
+  width: "48rem", // 270pt
+  height: "12rem", // 68pt
+  padding: "1.25rem 1.875rem", // 20pt 30pt
+  borderRadius: "3.75rem",
+  border: "4px solid #FFB45C",
+  backgroundColor: "rgba(255, 180, 92, 0.1)", // 10% opacity
+  fontFamily: "commuter_sans",
+  fontSize: "4rem",
+  color: "white",
+  display: "flex",
+  justifyContent: "center",
+  alignItems: "center",
+  boxShadow: "0px 0px 5px rgba(0, 0, 0, 0.3)",
+};
```
Lets create a .MyComp.module.scss file to define these styles (and the inline ones) and import the module (see other existing components for reference as this is the pattern through the codebase.  `MyComp` also needs a more descriptive name

chufucious replied:
Got it!

wileymc commented on public/images/video-generator/bill-status-became-law-darkbg.svg:
Ideally we won't have these images in the codebase. Will it impact performance if we just reference the code instead of the static image? If so, these should be saved in an s3 bucket and imported.

chufucious replied:
Got it, will move them to S3 once we get that set up- it shouldn't impact Remotion's performance. [Reference](https://www.remotion.dev/docs/assets#using-images)

wileymc reviewed (changes requested):
This is a good start! Left some comments mostly related to inline styles, naming, and general questions.

wileymc commented on pages/create-video/vercel.json:
Looks like this file no longer exists? De we need this command?

chufucious replied:
Will remove!

chufucious commented:
@wileymc 

Fixed build issues. Next didn't like `remotion` being inside the pages directory, so moved it up to `web` and renamed it `video` (Next didn't like it being named `remotion`)!
